src/coreclr/nativeaot/Runtime/EHHelpers.cpp

```diff
@@ -90,10 +90,15 @@ COOP_PINVOKE_HELPER(int32_t, RhGetModuleFileName, (HANDLE moduleHandle, _Out_ co
 
 COOP_PINVOKE_HELPER(void, RhpCopyContextFromExInfo, (void * pOSContext, int32_t cbOSContext, PAL_LIMITED_CONTEXT * pPalContext))
 {
-    UNREFERENCED_PARAMETER(cbOSContext);
     ASSERT((size_t)cbOSContext >= sizeof(CONTEXT));
+    memset(pOSContext, 0, cbOSContext);
     CONTEXT* pContext = (CONTEXT *)pOSContext;
+
+    // Fill in CONTEXT_CONTROL registers that were not captured in PAL_LIMITED_CONTEXT.
+    PopulateControlSegmentRegisters(pContext);
+
 #if defined(UNIX_AMD64_ABI)
+    pContext->ContextFlags = CONTEXT_CONTROL | CONTEXT_INTEGER;
```

Comment:
Since this happens for all the architectures, can you please do it just once before the #if above? And we could exclude the context zeroing / segment registers setting for WASM too.

Reply:
Done. I left the assert in for all platforms, due to this comment:

https://github.com/dotnet/runtime/blob/main/src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/ExceptionHandling.cs#L125